pkg/cli/webhook/webhook.go

```diff
@@ -102,3 +103,33 @@ func (w *Options) Install(ctx context.Context, providerType string) error {
 	// update repo cr with webhook secret
 	return w.updateRepositoryCR(ctx, response)
 }
+
+func GetProviderName(url string) (string, error) {
+	var (
+		err          error
+		providerName string
+	)
+	switch {
+	case strings.Contains(url, "github"):
+		providerName = "github"
+	case strings.Contains(url, "gitlab"):
+		providerName = "gitlab"
+	case strings.Contains(url, "bitbucket-cloud"):
+		providerName = "bitbucket-cloud"
+	default:
+		msg := "Please enter the provider name to setup the webhook:"
+		providerLabels := make([]string, 0, len(ProviderTypes))
+		for _, label := range ProviderTypes {
+			providerLabels = append(providerLabels, label)
```

Comment:
this is not goona change for each request so we can declare this as file level array?